jetty-io/src/main/java/org/eclipse/jetty/io/ssl/SslConnection.java

```diff
@@ -741,6 +741,18 @@ else if (_decryptedInput == null)
                                         throw new IllegalStateException("Unexpected unwrap result " + unwrapResultStatus);
 
                                     case BUFFER_UNDERFLOW:
+                                        // Continue if we can compact?
+                                        if (BufferUtil.compact(_encryptedInput))
+                                            continue;
+
+                                        // Are we out of space?
+                                        if (BufferUtil.space(_encryptedInput) == 0)
+                                        {
+                                            BufferUtil.clear(_encryptedInput);
+                                            throw new SSLHandshakeException("Encrypted buffer max length exceeded");
+                                        }
+
+                                        _underFlown = true;
```

Comment:
I'm concerned about the fall through to OK here.  I'm wondering if we should not just return like:
```suggestion
                                        _underFlown = true;
                                        break decryption;
```
This is the probably outcome from falling through anyway.

At the very least we need a comment saying that we intend to fall through and why.

Reply:
Added the comment.